src/App.vue

```diff
@@ -450,77 +431,60 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
     this.appReady = true
   }
 
-  /** Handle the emitted actions from the fee summary component.
-   * @param event The triggered event from the fee summary.
-   * */
-  private async handleSummaryActions (event: SummaryActions): Promise<void> {
-    switch (event) {
+  /**
+   * Handles actions from the fee summary component.
+   * NOTE: This is only implemented for Alteration filings atm.
+   * @param action the emitted action
+   */
+  private async handleFeeSummaryActions (action: SummaryActions): Promise<void> {
+    switch (action) {
       case SummaryActions.SAVE_RESUME_LATER:
-        // Save filing and return to dashboard
+        // Save filing and return to dashboard.
         await this.onClickSave()
-        this.goToDashboard(true)
+        this.goToDashboard()
         break
       case SummaryActions.DELETE_ALL:
-        this.goToDashboard()
+        this.$root.$emit('delete-all')
         break
       case SummaryActions.CONFIRM:
-        // If Summary Mode: Check validity, save and file else move into summary mode.
-        if (this.isSummaryMode) await this.validateApp()
-        else await this.validateComponents()
+        if (this.isSummaryMode) {
+          // Check validity, and if OK then save and file.
+          await this.validateApp()
+        } else {
+          // Check validity, and if OK then go to summary page.
+          await this.validateCompanyInfoPage()
+        }
         break
     }
   }
 
   /** Redirects to Manage Businesses dashboard. */
   private goToManageBusinessDashboard (): void {
     this.fileAndPayInvalidNameRequestDialog = false
+    this.setHaveUnsavedChanges(false)
+    // FUTURE: Manage Businesses URL should come from config
```

Comment:
If you see one of these FUTURE comments and you have time, go ahead and do it. (I'm out of time for this ticket.)